sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -608,6 +723,22 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
         val tz = ctx.addReferenceObj("timeZone", timeZone)
         (c, evPrim, evNull) => s"""$evPrim = UTF8String.fromString(
           org.apache.spark.sql.catalyst.util.DateTimeUtils.timestampToString($c, $tz));"""
+      case ar: ArrayType =>
+        val bufferClass = classOf[StringBuffer].getName
+        val buffer = ctx.addMutableState(bufferClass, "buffer", v => s"$v = new $bufferClass();")
+        val writeArrayToBuffer = codegenWriteArrayToBuffer(ar, buffer, ctx)
+        val arrayToStringCode =
+          s"""
+             |if (!${ev.isNull}) {
```

Comment:
why is it? We are in `nullSafeCastFunction`, and ideally we don't need the `ev: ExprCode` parameter.

Reply:
ya, I missed. Dropped `ev` now.